jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.c

```diff
@@ -63,156 +63,156 @@ ecma_builtin_regexp_prototype_compile (ecma_value_t this_arg, /**< this argument
 
-        if (ecma_is_value_true (multiline_value))
-        {
-          flags |= RE_FLAG_MULTILINE;
-        }
+    if (ecma_is_value_true (multiline_value))
+    {
+      flags |= RE_FLAG_MULTILINE;
+    }
 
-        ECMA_TRY_CATCH (obj_this, ecma_op_to_object (this_arg), ret_value);
-        ecma_object_t *this_obj_p = ecma_get_object_from_value (obj_this);
+    ecma_value_t obj_this = ecma_op_to_object (this_arg);
+    if (ECMA_IS_VALUE_ERROR (obj_this))
+    {
+      return obj_this;
+    }
+    ecma_object_t *this_obj_p = ecma_get_object_from_value (obj_this);
 
-        /* Get bytecode property. */
-        ecma_value_t *bc_prop_p = &(((ecma_extended_object_t *) this_obj_p)->u.class_prop.u.value);
+    /* Get bytecode property. */
+    ecma_value_t *bc_prop_p = &(((ecma_extended_object_t *) this_obj_p)->u.class_prop.u.value);
 
-        /* TODO: We currently have to re-compile the bytecode, because
-         * we can't copy it without knowing its length. */
-        const re_compiled_code_t *new_bc_p = NULL;
-        ecma_value_t bc_comp = re_compile_bytecode (&new_bc_p, pattern_string_p, flags);
-        /* Should always succeed, since we're compiling from a source that has been compiled previously. */
-        JERRY_ASSERT (ecma_is_value_empty (bc_comp));
+    /* TODO: We currently have to re-compile the bytecode, because
+     * we can't copy it without knowing its length. */
+    const re_compiled_code_t *new_bc_p = NULL;
+    ecma_value_t bc_comp = re_compile_bytecode (&new_bc_p, pattern_string_p, flags);
+    /* Should always succeed, since we're compiling from a source that has been compiled previously. */
+    JERRY_ASSERT (ecma_is_value_empty (bc_comp));
 
-        ecma_deref_ecma_string (pattern_string_p);
+    ecma_deref_ecma_string (pattern_string_p);
 
-        re_compiled_code_t *old_bc_p = ECMA_GET_INTERNAL_VALUE_ANY_POINTER (re_compiled_code_t, *bc_prop_p);
+    re_compiled_code_t *old_bc_p = ECMA_GET_INTERNAL_VALUE_ANY_POINTER (re_compiled_code_t, *bc_prop_p);
 
-        if (old_bc_p != NULL)
-        {
-          /* Free the old bytecode */
-          ecma_bytecode_deref ((ecma_compiled_code_t *) old_bc_p);
-        }
+    if (old_bc_p != NULL)
+    {
+      /* Free the old bytecode */
+      ecma_bytecode_deref ((ecma_compiled_code_t *) old_bc_p);
+    }
+
+    ECMA_SET_INTERNAL_VALUE_POINTER (*bc_prop_p, new_bc_p);
 
-        ECMA_SET_INTERNAL_VALUE_POINTER (*bc_prop_p, new_bc_p);
+    re_initialize_props (this_obj_p, pattern_string_p, flags);
+    ecma_free_value (obj_this);
+
+    return ECMA_VALUE_UNDEFINED;
+  }
 
-        re_initialize_props (this_obj_p, pattern_string_p, flags);
+  ecma_string_t *pattern_string_p = NULL;
 
-        ret_value = ECMA_VALUE_UNDEFINED;
+  /* Get source string. */
+  ecma_value_t ret_value = ecma_regexp_read_pattern_str_helper (pattern_arg, &pattern_string_p);
+  if (ECMA_IS_VALUE_ERROR (ret_value))
+  {
+    JERRY_ASSERT (pattern_string_p == NULL);
+    return ret_value;
+  }
```

Comment:
I'm not completely sure but my gut feeling tells me that an assert would be good to have here. `ret_value` is not used later on anywhere. However, it is an `ecma_value_t`, which is usually something that has to be released unless it is known that is of type that needs no release. That is, if `ret_value` is not an error then it must be undefined(?)

Reply:
It must be empty if not an error. I can add an assertion for that.